traits/ctraits.c

```diff
@@ -942,24 +941,6 @@ has_traits_init ( PyObject * obj, PyObject * args, PyObject * kwds ) {
         Py_DECREF( value );
     }
 
-    /* Notify any interested monitors that a new object has been created: */
-    for ( i = 0, n = PyList_GET_SIZE( _HasTraits_monitors ); i < n; i++ ) {
-        value = PyList_GET_ITEM( _HasTraits_monitors, i );
-        assert( PyTuple_Check( value ) );
-        assert( PyTuple_GET_SIZE( value ) == 2 );
-
-        klass   = PyTuple_GET_ITEM( value, 0 );
-        handler = PyTuple_GET_ITEM( value, 1 );
-
-        if ( PyObject_IsInstance( obj, klass ) > 0 ) {
-            handler_args = PyTuple_New( 1 );
-            PyTuple_SetItem( handler_args, 0, obj );
-            Py_INCREF( obj );
-            PyObject_Call( handler, handler_args, NULL );
```

Comment:
This is also unsafe: we're calling arbitrary code that could raise an exception, and not checking the return value.